doc/developer-guide/developer-guide.rst

```diff
@@ -0,0 +1,164 @@
+.. _Developer_Guide:
+
+#########################################
+DSM Developer Guide
+#########################################
+
+The DSM is a non-configurable CSC that looks for output telemetry files from 
+the Dome Seeing Monitor UI and sends that information across SAL for recording
+in the EFD. The CSC also has two simulation modes in addition to the standard
+operational (real) mode and they will be described later. 
+
+.. _Dependencies:
+
+Dependencies
+============
+
+* `SAL <https://ts-sal.lsst.io>`_ - 5.0.0
+* ts_xml - v6.1.0
+* ts_salobj - v6.0.0
+* `aionotify <https://github.com/rbarrois/aionotify>`_ - 0.2.0
+
+.. Linking to the previous versions may also be worthwhile, depending on the CSC
+
+.. _API:
+
+DSM API
+=============================
+
+The content in this section is autogenerated from docstrings.
+
+.. The code below should insert the docstrings from the code.
+
+.. automodapi:: lsst.ts.dsm
+    :no-main-docstr:
+    :no-inheritance-diagram:
+
+
+.. _Build:
+
+Build and Test
+==============
+
+The recommended method for development is to get the T&S development Docker container
+(need doc link). Since the DSM code is not shipped with the development container, a
+clone of the code is necessary. Determine a place on your computer and get it via this
+
+.. prompt:: bash
+
+  git clone git@github.com:lsst-ts/ts_dsm.git
+
+It is recommended that the code be cloned into another directory as the potential
+of needing other repositories is possible. Change or create the appropriate branch in the clone. Remember the location housing the repository.
```

Comment:
I find this recommendation rather confusing. I think you are saying that you should clone the repo into a directory that you use for other repositories. Please either reword or consider deleting it (this is standard git practice).

My personal recommendation is to clone into a directory that you can mount in your Docker image. But that's all explained in the docs on how to use these Docker development images. Best not to go into that here.